src/movies.js

```diff
@@ -1,16 +1,149 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(array) {
+  let directorsArr = [];
+  array.map((movie) => {
+    directorsArr.push(movie.director);
+    return movie.director;
+  });
+  return directorsArr;
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+// One of the most famous directors in cinema is Steven Spielberg, and he has some really awesome drama movies that are on our list, but we want to know how many of them are there.
+// Go ahead and create a howManyMovies() function that receives an array as a parameter and filter 👀 the array so we can have only the drama movies where Steven Spielberg is the director.
+
+function howManyMovies(array) {
+  if (array.length === 0) return 0;
+  const filteredMovies = array.filter((movie) => {
+    return (
+      movie.director === "Steven Spielberg" && movie.genre.includes("Drama")
+    );
+  });
+  return filteredMovies.length;
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+// These are the best movies based on their rates, so supposedly all of them have a remarkable rate. In this iteration, we want to know the average rate of all of them and display it on the console. Create a ratesAverage() function that receives an array as a parameter and solves the challenge.
+
+// The rate must be returned rounded to 2 decimals!
+
+// 💡 Maybe you want to "reduce" the data to a single value. 😉
+
+function ratesAverage(array) {
+  if (array.length === 0) return 0;
+  const sum = array.reduce((acc, val) => {
+    if (val.rate === "" || "rate" in val === false) return acc;
+    return acc + val.rate;
+  }, 0);
+  return Number((sum / array.length).toFixed(2));
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+// Drama is the genre that repeats the most on our array. Apparently, people love drama! 👀
+// Create a dramaMoviesRate() function that receives an array as a
+//parameter to get the average rate of all drama movies! Let's see if it is better than the general average.
+
+function dramaMoviesRate(array) {
+  const sum = array.reduce((acc, val) => {
+    if (val.genre.includes("Drama")) {
+      return acc + val.rate;
+    } else {
+      return acc;
+    }
+  }, 0);
+
+  return Number((sum / array.length).toFixed(2));
+}
+
+//another solution:
+// function dramaMoviesRate(movies) {
+//   const dramas = movies.filter(function (movie) {
+//       return movie.genre.indexOf('Drama') !== - 1
+//   })
+//   return ratesAverage(dramas);
+// }
```

Comment:
Hello Vesna! Great work! I noticed you commented out a super cool way to do the dramaMoviesRate function. Try it out!